docs/export_kernels.rst

```diff
@@ -0,0 +1,217 @@
+Export Taichi kernels to C source
+=================================
+
+The C backend of Taichi allows you to **export Taichi kernels to C source**.
+
+The exported Taichi program, which are simply C99 compatible source,
+does not necessary to be launched from Python. Instead, you may use them in
+your C/C++ project. Or even calling them in Javascript via Empscripten.
+
+Each C function corresponds to a Taichi kernel.
+For example, ``Tk_init_c6_0()`` may corresponds to ``init()`` in ``mpm88.py``.
+
+Only the source is need when copying, all the required Taichi runtimes are
+included in that single C source file.
+
+This also allows commercial people to distribute their Taichi program in
+binary format by linking this file with their project.
+
+.. note::
+
+    Currently this feature is only officially supported on the C backend.
+    Also note that the C backend is only released on **Linux** platform for now.
+
+
+The workflow of export
+----------------------
+
+Use ``ti.core.start_recording`` in the Taichi program you want to export.
+
+Suppose you want to export the `examples/mpm88.py <https://github.com/taichi-dev/taichi/blob/master/examples/mpm88.py>`_, here are the workflow:
+
+Export YAML
++++++++++++
+
+First, modify the ``mpm88.py`` a little bit:
+
+.. code-block:: python
+
+    import taichi as ti
+
+    ti.core.start_recording('mpm88.yml')
+    ti.init(arch=ti.cc)
+
+    ... # your program
+
+And run it. Close the GUI window once particles are shown up correctly.
+
+This will save all the kernels in ``mpm88.py`` to ``mpm88.yml``:
+
+.. code-block:: yaml
+
+   - action: "compile_kernel"
+      kernel_name: "init_c6_0"
+      kernel_source: "void Tk_init_c6_0(struct Ti_Context *ti_ctx) {\n  for (Ti_i32 tmp0 = 0; tmp0 < 8192...\n"
+    - action: "launch_kernel"
+      kernel_name: "init_c6_0"
+    ...
+
+.. note::
+
+    Equivalently, you may also specify these two arguments from environment
+    variables if you are on Unix-alike system:
```

Comment:
```suggestion
    variables on Unix-like system:
```
Unix-like is a better terminology, as (!wiki)[https://en.wikipedia.org/wiki/Unix-like] showed IMO